http/src/lib.rs

```diff
@@ -565,7 +565,10 @@ fn serve<M: jsonrpc::Metadata, S: jsonrpc::Middleware<M>>(
 				}))
 				.map_err(|_| ())
 		})
-		.and_then(|_| done_tx.send(()))
+		.and_then(|(_, other)| {
+			drop(other);
```

Comment:
Could you add a comment what are we dropping here and why? Something like:

"We drop the `Server` first to prevent a situation where main thread terminates before the `Server` is properly dropped (see #504 for more details)".

Reply:
maybe s/other/server/ then?